### PR TITLE
kubelet dra: add lock to addCDIDevices

### PR DESCRIPTION
**Pull request by moshe010**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/cm/dra/claiminfo.go [hunk 1]
```diff
@@ -36,18 +36,39 @@ type ClaimInfo struct {
...
+	info.Lock()
+	defer info.Unlock()
 
-	res.PodUIDs.Delete(string(podUID))
+	info.PodUIDs.Delete(string(podUID))
+}
+
+func (info *ClaimInfo) addCDIDevices(pluginName string, cdiDevices []string) error {
+	info.Lock()
+	defer info.Unlock()
```
Comment (klueska) [inline pkg/kubelet/cm/dra/claiminfo.go:55]: nit, put a newline after this.

**Comment by ffromani** [comment]
@klueska @moshe010 any more context about this PR?

**Comment by ffromani** [comment]
/priority important-soon

considering the recent activity in this area and the nature of the PR

File: pkg/kubelet/cm/dra/claiminfo.go [hunk 2]
```diff
@@ -36,18 +36,40 @@ type ClaimInfo struct {
...
+	// to get CDI device names from the ContainerConfig.CDIDevices field
+	annotations, err := generateCDIAnnotations(info.ClaimUID, info.DriverName, cdiDevices)
+	if err != nil {
+		return fmt.Errorf("failed to generate container annotations, err: %+v", err)
+	}
+
+	if info.CDIDevices == nil {
+		info.CDIDevices = make(map[string][]string)
+	}
+
```
Comment (bart0sh) [inline pkg/kubelet/cm/dra/claiminfo.go:68]: Would it make sense to lock here, just before the code that changes info?

**Comment by bart0sh** [comment]
@moshe010 There seems to be a code that reads annotations and CDIDevices. It should be also locked, I believe.

**Comment by moshe010** [comment]
> @klueska @moshe010 any more context about this PR?

It just something that we missed when we merged https://github.com/kubernetes/kubernetes/pull/116513. 

We spot it after merge.

**Pull request merged**